pyflakes/checker.py

```diff
@@ -870,6 +922,38 @@ def ignore(self, node):
     # additional node types
     COMPREHENSION = KEYWORD = FORMATTEDVALUE = handleChildren
 
+    def DICT(self, node):
+        # Complain if there are duplicate keys with different values
+        # If they have the same value it's not going to cause potentially
+        # unexpected behaviour so we'll not complain.
```

Comment:
From a very quick and dirty test (timing 10000 runs of a func which does result = [str(key) for key in item.keys()] with no gating, a test for == {} and a test for len(thing) == 0), it seems like the list comprehension finishes just as fast as the gated versions (possibly faster actually, though that may be an artefact of the way the test was constructed).

Probably not worth it given that it gives no obvious gain in efficiency anyway.